lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,293 @@
+    def send_activation_email(request):
+        """
+        Send activation email with reset password.
+        """
+        form = PasswordResetFormNoActive(request.data)
+        if form.is_valid():
+            form.save(
+                use_https=request.is_secure(),
+                from_email=configuration_helpers.get_value(
+                    'email_from_address', settings.DEFAULT_FROM_EMAIL
+                ),
+                request=request,
+                subject_template_name='sga_api/set_password_subject.txt',
+                email_template_name='sga_api/set_password_email.html'
+            )
+
+
+class BulkEnrollView(APIView, ApiKeyPermissionMixIn):
+    """
+    API endpoint for bulk enrolling/unenrolling users on the courses.
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    def post(self, request):
+        """
+        Bulk enrolling/unenrolling users on the courses.
+
+        Enroll or unenroll users on the courses.
+        """
+        data = request.data
+        users = data.get('users')
+        courses = data.get('courses')
+        action = data.get('action', 'enroll')
+        email_students = data.get('email_students', False)
+        auto_enroll = data.get('auto_enroll', False)
+
+        if action not in ['enroll', 'unenroll']:
+            error_msg = "action must be enroll or unenroll"
+        elif not isinstance(courses, (list, tuple)):
+            error_msg = "courses must be a list of courses id "
+        elif not isinstance(users, (list, tuple)):
+            error_msg =  "users must be a list of identifiers(username or email)"
+        else:
+            error_msg = None
+
+        if error_msg:
+            return Response(
+                data={"error_message": error_msg},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            result = self._enroll_unenrol_users_for_courses(
+                courses, users, request, action, email_students, auto_enroll
+            )
+
+            return Response(
+                data={
+                    'action': action,
+                    'courses': result,
+                    'email_students':email_students,
+                    'auto_enroll': auto_enroll,
+                },
+                status=status.HTTP_200_OK
+            )
+        except Exception as e:
+            log.error(e.message)
+            return Response(
+                data={"error_message": e.message},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+    def _enroll_unenrol_users_for_courses(
+        self, list_of_courses_id, list_of_users_email,
+        request, action='enroll', email_students=False, auto_enroll=False
+    ):
+        courses = {}
+        for course_id in list_of_courses_id:
+            results = self._enroll_unenrol_users_for_course(
+                course_id, list_of_users_email, request,
```

Comment:
Two strings for function's parameters is not an option please do as following: 
```
results = self._enroll_unenrol_users_for_course(
                course_id, list_of_users_email, request,
)
```
or
```
results = self._enroll_unenrol_users_for_course(
    course_id, 
    list_of_users_email, 
    request,
)
```

Reply:
I did them by  this page form edx documentation:
https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines

I made a recommendation on the account settings editor and configured my editor according to the requirements on the same page.